WebRTCPlugin/WebRTCProxy.cpp

```diff
@@ -8,321 +7,319 @@
+	if (!pci)
+		//Error
+		return E_INVALIDARG;
+
+	//Set event thread
+	pc->SetThread(eventThread);
+
+	//Attach to PC
+	pc->Attach(pci);
+
+	//Get Reference to pass it to JS
+	*peerConnection = pc->GetUnknown();
+
+	//Add JS reference
+	(*peerConnection)->AddRef();
+
+	//OK
+	return hresult;
 }
 
 
 STDMETHODIMP WebRTCProxy::createLocalAudioTrack(VARIANT constraints, IUnknown** track)
 {
-  const cricket::AudioOptions options;
-  //Create audio source
-  auto audioSource = peer_connection_factory_->CreateAudioSource(options);
+	const cricket::AudioOptions options;
+	//Create audio source
+	auto audioSource = peer_connection_factory_->CreateAudioSource(options);
+
+	//Ensure it is created
+	if (!audioSource)
+		return E_UNEXPECTED;
 
-  //Ensure it is created
-  if (!audioSource)
-    return E_UNEXPECTED;
+	//Create track
+	rtc::scoped_refptr<webrtc::MediaStreamTrackInterface> audioTrack = peer_connection_factory_->CreateAudioTrack("audio", audioSource);
 
-  //Create track
-  rtc::scoped_refptr<webrtc::MediaStreamTrackInterface> audioTrack = peer_connection_factory_->CreateAudioTrack("audio", audioSource);
+	//Ensure it is created
+	if (!audioTrack)
+		return E_UNEXPECTED;
 
-  //Ensure it is created
-  if (!audioTrack)
-    return E_UNEXPECTED;
+	//Create activeX object which is a
+	CComObject<MediaStreamTrack>* mediaStreamTrack;
+	HRESULT hresult = CComObject<MediaStreamTrack>::CreateInstance(&mediaStreamTrack);
 
-  //Create activeX object which is a
-  CComObject<MediaStreamTrack>* mediaStreamTrack;
-  HRESULT hresult = CComObject<MediaStreamTrack>::CreateInstance(&mediaStreamTrack);
+	if (FAILED(hresult))
+		return hresult;
 
-  if (FAILED(hresult))
-    return hresult;
-  
-  //Attach to native track
-  mediaStreamTrack->Attach(audioTrack);
+	//Attach to native track
+	mediaStreamTrack->Attach(audioTrack);
 
-  //Set dummy audio  label
-  mediaStreamTrack->SetLabel("Default Audio Device");
+	//Set dummy audio  label
+	mediaStreamTrack->SetLabel("Default Audio Device");
 
-  //Get Reference to pass it to JS
-  *track = mediaStreamTrack->GetUnknown();
+	//Get Reference to pass it to JS
+	*track = mediaStreamTrack->GetUnknown();
 
-  //Add JS reference
-  (*track)->AddRef();
+	//Add JS reference
+	(*track)->AddRef();
 
-  //OK
-  return hresult;
+	//OK
+	return hresult;
 }
 
+class CapturerTrackSource : public webrtc::VideoTrackSource
+{
+public:
+	static rtc::scoped_refptr<CapturerTrackSource> Create()
+	{
+		const size_t kWidth = 640;
+		const size_t kHeight = 480;
+		const size_t kFps = 30;
+		const size_t kDeviceIndex = 0;
+		std::unique_ptr<VcmCapturer> capturer =
+			absl::WrapUnique(VcmCapturer::Create(kWidth, kHeight, kFps, kDeviceIndex));
+		if (!capturer)
+		{
+			return nullptr;
+		}
+		return new rtc::RefCountedObject<CapturerTrackSource>(std::move(capturer));
+	}
+
+protected:
+	explicit CapturerTrackSource(std::unique_ptr<VcmCapturer> capturer)
+		: VideoTrackSource(/*remote=*/false), capturer(std::move(capturer))

```

Comment:
meaning the plugin will only support subscribing to video, and not publishing?

Reply:
I don't see where this implication is made. I think you'd have to look at this in combination with the old source code, then you can modify it to suit your intended purposes.